mm2src/coins/lightning/ln_conf.rs

```diff
@@ -129,10 +102,50 @@ pub struct OurChannelsConfig {
     /// our real on-chain channel UTXO in each invoice and requiring that our counterparty only
     /// relay HTLCs to us using the channel's SCID alias.
     pub negotiate_scid_privacy: Option<bool>,
+    /// Sets the percentage of the channel value we will cap the total value of outstanding inbound
+    /// HTLCs to.
+    pub max_inbound_in_flight_htlc_percent: Option<u8>,
+    /// Set to announce the channel publicly and notify all nodes that they can route via this
+    /// channel.
+    pub announced_channel: Option<bool>,
+    /// When set, we commit to an upfront shutdown_pubkey at channel open.
+    pub commit_upfront_shutdown_pubkey: Option<bool>,
 }
 
-impl From<OurChannelsConfig> for ChannelHandshakeConfig {
-    fn from(config: OurChannelsConfig) -> Self {
+impl OurChannelsConfigs {
+    pub fn update(&mut self, config: OurChannelsConfigs) {
+        if let Some(confs) = config.inbound_channels_confirmations {
+            self.inbound_channels_confirmations = Some(confs);
+        }
+
+        if let Some(delay) = config.counterparty_locktime {
+            self.counterparty_locktime = Some(delay);
+        }
+
+        if let Some(min) = config.our_htlc_minimum_msat {
+            self.our_htlc_minimum_msat = Some(min);
+        }
+
+        if let Some(scid_privacy) = config.negotiate_scid_privacy {
+            self.negotiate_scid_privacy = Some(scid_privacy);
+        }
+
+        if let Some(max_inbound_htlc) = config.max_inbound_in_flight_htlc_percent {
+            self.max_inbound_in_flight_htlc_percent = Some(max_inbound_htlc);
+        }
+
+        if let Some(announce) = config.announced_channel {
+            self.announced_channel = Some(announce);
+        }
+
+        if let Some(commit) = config.commit_upfront_shutdown_pubkey {
+            self.commit_upfront_shutdown_pubkey = Some(commit);
+        }
```

Comment:
Referencing this https://github.com/KomodoPlatform/atomicDEX-API/pull/1452#discussion_r958797083, If a field is not provided in `channel_options` or `channel_configs` in `OpenChannelRequest` we get it from config by applying `update` to `channel_options` and `channel_configs` https://github.com/KomodoPlatform/atomicDEX-API/blob/6d6ab4e0d3dc40bea6dbbeb3f5c795f7e265f8b3/mm2src/coins/lightning.rs#L906-L919 As you can see, we get a clone from config and we update only the values that are provided in the request, leaving the other values as is, instead of equaling it to `None` which would happen if I used the 2 examples you proposed.

I think maybe this function should be renamed to better reflect what it does, `get_missing_values_from_conf` or something in this regard can be a better name for it, but `update` is also not a bad name since what the function does is that it updates the config with only the values that are provided. What do you think, do you have a better name for this?